internal/pkg/keeper/client.go

```diff
@@ -140,9 +143,80 @@ func (client *keeperClient) GetConfiguration(configStruct interface{}) (interfac
 }
 
 func (client *keeperClient) WatchForChanges(updateChannel chan<- interface{}, errorChannel chan<- error, configuration interface{}, waitKey string) {
+	// get the service configuration
+	config, err := client.GetConfiguration(&models.ConfigurationStruct{})
+	if err != nil {
+		errorChannel <- err
+		return
+	}
+
+	messages := make(chan msgTypes.MessageEnvelope)
+	topic := path.Join(keeperTopicPrefix, client.configBasePath, waitKey, "#")
+	topics := []msgTypes.TopicChannel{
+		{
+			Topic:    topic,
+			Messages: messages,
+		},
+	}
+	msgBusConfig := config.(*models.ConfigurationStruct).MessageQueue
+	messageBus, err := messaging.NewMessageClient(msgTypes.MessageBusConfig{
+		SubscribeHost: msgTypes.HostInfo{
+			Host:     msgBusConfig.Host,
+			Port:     msgBusConfig.Port,
+			Protocol: msgBusConfig.Protocol,
+		},
+		Type:     msgBusConfig.Type,
+		Optional: msgBusConfig.Optional,
```

Comment:
add a deepCopy func to produce a copy of messageBusConfig, and add a random number suffix in the MqttClientId option.
reference:
https://github.com/IOTechSystems/edgex-go-private/pull/314/files
https://github.com/IOTechSystems/edgex-go-private/blob/b95546cea1628225578aae0059b0751de3f401c8/internal/support/sparkplug/mqtt/client.go#L249-L252

Reply:
Random number suffix in the MqttClientId added, thanks.